pilot/cmd/pilot-discovery/main.go

```diff
@@ -116,6 +116,9 @@ func init() {
 	serverArgs.MCPCredentialOptions.AttachCobraFlags(discoveryCmd)
 
 	// Config Controller options
+	discoveryCmd.PersistentFlags().BoolVar(&serverArgs.Config.DisableInstallCRDs, "disable-install-crds", false,
```

Comment:
I would just remove the verification/creation completely. It was needed before we had the helm installer and 
for local debugging. 

This removes complexity.

Reply:
It turns out it isn't quite that simple.  Based on my original testing, some of the integration tests still rely on this CRD creation to exist (this is why the functionality is enabled by default).  It does seem like a reasonable clean-up task to completely remove the CRD creation functionality altogether, but this PR is focused on providing a mechanism to prevent the risky behavior without causing a potentially backwards incompatible change.